File: Arrays/Kadanes_Algorithm.py
class Solution:
    # Function to find the sum of contiguous subarray with maximum sum.
    def maxSubArraySum(self, arr, n):
        ##Your code here
        curr = 0
        ans = float('-inf')
        for i in range (n):
            curr+=arr[i]
            ans = max(curr,ans)
            if curr < 0:
                curr = 0
        return ans

File: Arrays/Kadanes_Algorithm.cpp
class Solution {
  public:
    long long maxSubarraySum(int arr[], int n) {

        // Your code here
        long long curr = 0;
        long long ans = INT_MIN;
        
        for(int i=0;i<n;i++){
            curr+=arr[i];
            ans = max(curr,ans);
            if(curr<0){
                curr = 0;
            }
        }
        return ans;
    }
};
